stubs/Flask-SocketIO/flask_socketio/namespace.pyi

```diff
@@ -0,0 +1,46 @@
+from _typeshed import Incomplete
+from typing import Any, Callable, TypedDict, TypeVar
+from typing_extensions import TypeAlias
+
+from socketio import Namespace as _Namespace
+
+from .socketio import SocketIO
+
+_T = TypeVar('T')
+_Json: TypeAlias = dict[str, _Json] | list[_Json] | int | str | float | bool | None
+
+class _Acks(TypedDict):
+    args: Any
+    namespace: Any
+
+class Namespace(_Namespace):
+    acks: Acks | None
+    socketio: SocketIO | None
+
+    def __init__(self, namespace=None) -> None: ...
+    def trigger_event(self, event: Incomplete, *args): ...
+    def emit(
+        self,
+        event: Incomplete,
+        data: Incomplete | None = None,
+        room: str | None = None,
+        include_self: bool = True,
+        namespace: str | None = None,
+        callback: Callable[..., _T] | None = None,
+    ) -> _T | tuple[str, int]: ...
+    def send(
+        self,
+        data: str | list[_Json] | dict[str, _Json],
+        room: str | None = None,
+        include_self: bool = True,
+        namespace=None,
+        callback: Callable | None = None,
+    ) -> None: ...
+    def close_room(self, room: str, namespace=None) -> None: ...
```

Comment:
missing type for `namespace`

Reply:
FWIW, while it's obviously good to have as many annotations as possible, we don't require stubs to be fully complete before accepting them to typeshed — we're happy to accept an initial PR with incomplete annotations, and for those to be improved in followup PRs. In fact, it makes it easier for us to review if it's done that way.

I'd advise trying to fix the CI failures first before worrying about whether or not the type annotations are 100% complete yet

Reply:
@miguelgrinberg Added type `namespace: str | None = None`.

@AlexWaygood Could you clarify regarding the CI tests? These seem to be mostly linter checks (eg. `pylint`) - so basically I have to correct my formatting somewhere?